truffle/src/main/java/org/jruby/truffle/runtime/object/ObjectGraph.java

```diff
@@ -133,8 +161,11 @@ private void visitObject(Object object, ObjectGraphVisitor visitor) throws StopV
             visitObject(entry.getKey(), visitor);
             visitObject(entry.getValue(), visitor);
             visitObject(entry.getNextInLookup(), visitor);
-        } else if (object instanceof MaterializedFrame) {
-            visitFrame((MaterializedFrame) object, visitor);
+        } else if (object instanceof Frame) {
+            visitFrame((Frame) object, visitor);
+        } else if (object instanceof InternalMethod) {
+            final InternalMethod method = (InternalMethod) object;
+            visitObject(method.getDeclarationFrame(), visitor);
```

Comment:
This is only needed in the case of a method defined by `define_method` with a proc, right?